hdt_arm:
  # Publish all joint states ----------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------
  arm_controller:
    # type: effort_controllers/JointPositionController
    type: effort_controllers/JointTrajectoryController
    joints: 
      - drive1_joint
      - drive2_joint
      - drive3_joint
      - drive4_joint
      - drive5_joint
      - drive6_joint
    gains:
      drive1_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}
      drive2_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}
      drive3_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}
      drive4_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}
      drive5_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}
      drive6_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}

  hand_controller:
    # type: effort_controllers/JointPositionController
    type: effort_controllers/JointTrajectoryController
    joints:
      - pincer_joint
      - pincer_joint2
    gains:
      pincer_joint: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}
      pincer_joint2: {p: 1000.0, i: 0.01, d: 0.1, i_clamp: 0.0}


# joint_group_position_controller:
#   type: position_controllers/JointGroupPositionController
#   joints:
#     - drive1_joint
#     - drive2_joint
#     - drive3_joint
#     - drive4_joint
#     - drive5_joint
#     - drive6_joint

# arm_controller:
#   type: position_controllers/JointTrajectoryController
#   joints:
#     - drive1_joint
#     - drive2_joint
#     - drive3_joint
#     - drive4_joint
#     - drive5_joint
#     - drive6_joint
#   gains:
#     drive1_joint:
#       p: 100
#       d: 1
#       i: 1
#       i_clamp: 1
#     drive2_joint:
#       p: 100
#       d: 1
#       i: 1
#       i_clamp: 1
#     drive3_joint:
#       p: 100
#       d: 1
#       i: 1
#       i_clamp: 1
#     drive4_joint:
#       p: 100
#       d: 1
#       i: 1
#       i_clamp: 1
#     drive5_joint:
#       p: 100
#       d: 1
#       i: 1
#       i_clamp: 1
#     drive6_joint:
#       p: 100
#       d: 1
#       i: 1
#       i_clamp: 1
